lapack/testlapack/general.go

```diff
@@ -1068,15 +1068,15 @@ func extract2x2Block(t []float64, ldt int) (a, b, c, d float64) {
 
 // isSchurCanonical returns whether the 2×2 matrix [a b; c d] is in Schur
 // canonical form.
-func isSchurCanonical(a, b, c, d float64) bool {
-	return c == 0 || (b != 0 && a == d && math.Signbit(b) != math.Signbit(c))
+func isSchurCanonical(a, b, c, d, tol float64) bool {
+	return math.Abs(c) <= tol || (b != 0 && math.Abs(a-d) <= tol && math.Signbit(b) != math.Signbit(c))
```

Comment:
Why is the tolerance needed?

Reply:
Can't recall why, it seems like it is not necessary anymore.